tests/test_c_api.cc

```diff
@@ -0,0 +1,176 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation
+ * nor does it submit to any jurisdiction.
+ */
+
+/// @file   test_c_api.cc
+/// @date   Dec 2024
+/// @author Christopher Bradley
+
+#include "metkit/api/metkit_c.h"
+#include "eckit/testing/Test.h"
+#include "eckit/types/Date.h"
+#include <cstring>
+
+using namespace eckit::testing;
+namespace metkit::test {
+
+// Wrapper around the C function calls that will throw an exception if the function fails
+// i.e. if the return value is not METKIT_SUCCESS
+#define METKIT_TEST_C(expr) \
```

Comment:
Function-like macro 'METKIT_TEST_C' used; consider a 'constexpr' template function

Reply:
This really really really should not be a macro :)

Reply:
My brain was in C mode